VN300_L554ZEQ/Core/Src/main.cpp

```diff
@@ -0,0 +1,621 @@
+  */
+/* USER CODE END Header */
+/* Includes ------------------------------------------------------------------*/
+#include "main.h"
+
+/* Private includes ----------------------------------------------------------*/
+/* USER CODE BEGIN Includes */
+#include "LOS_D_vn300.hpp"
+#include <string.h>
+#include <stdio.h>
+/* USER CODE END Includes */
+
+/* Private typedef -----------------------------------------------------------*/
+/* USER CODE BEGIN PTD */
+
+/* USER CODE END PTD */
+
+/* Private define ------------------------------------------------------------*/
+/* USER CODE BEGIN PD */
+#define MSG_LEN 2000
+/* USER CODE END PD */
+
+/* Private macro -------------------------------------------------------------*/
+/* USER CODE BEGIN PM */
+
+/* USER CODE END PM */
+
+/* Private variables ---------------------------------------------------------*/
+ADC_HandleTypeDef hadc1;
+
+UART_HandleTypeDef hlpuart1;
+UART_HandleTypeDef huart2;
+
+RTC_HandleTypeDef hrtc;
+
+PCD_HandleTypeDef hpcd_USB_FS;
+
+/* USER CODE BEGIN PV */
+
+/* USER CODE END PV */
+
+/* Private function prototypes -----------------------------------------------*/
+void SystemClock_Config(void);
+static void MX_GPIO_Init(void);
+static void MX_ADC1_Init(void);
+static void MX_LPUART1_UART_Init(void);
+static void MX_RTC_Init(void);
+static void MX_UCPD1_Init(void);
+static void MX_USB_PCD_Init(void);
+static void MX_USART2_UART_Init(void);
+static void MX_ICACHE_Init(void);
+/* USER CODE BEGIN PFP */
+
+/* USER CODE END PFP */
+
+/* Private user code ---------------------------------------------------------*/
+/* USER CODE BEGIN 0 */
+
+/* USER CODE END 0 */
+
+/**
+  * @brief  The application entry point.
+  * @retval int
+  */
+int main(void)
+{
+  /* USER CODE BEGIN 1 */
+
+  /* USER CODE END 1 */
+
+  /* MCU Configuration--------------------------------------------------------*/
+
+  /* Reset of all peripherals, Initializes the Flash interface and the Systick. */
+  HAL_Init();
+
+  /* USER CODE BEGIN Init */
+
+  /* USER CODE END Init */
+
+  /* Configure the system clock */
+  SystemClock_Config();
+
+  /* USER CODE BEGIN SysInit */
+
+  /* USER CODE END SysInit */
+
+  /* Initialize all configured peripherals */
+  MX_GPIO_Init();
+  MX_ADC1_Init();
+  MX_LPUART1_UART_Init();
+  MX_RTC_Init();
+  MX_UCPD1_Init();
+  MX_USB_PCD_Init();
+  MX_USART2_UART_Init();
+  MX_ICACHE_Init();
+  /* USER CODE BEGIN 2 */
+  const char* serial_cmd = "$VNWRG,06,0*XX";
+  const char* start_str = "START\r\n";
+  uint16_t j = 0;
+  char buffer[256];
+  memset(buffer, 0, 256);
+  char msg[MSG_LEN];
+  memset(msg, 0, MSG_LEN);
+
+  /* Set serial port */
+  HAL_UART_Transmit(&huart2, (uint8_t*)serial_cmd, strlen(serial_cmd), 1000);
+  HAL_UART_Transmit(&hlpuart1, (uint8_t*)start_str, strlen(start_str), 100);
+
+  /* Driver Init */
+  static Position *vn300obj = &VN300::getInstance();
+
+  requestData_t requestData = {true, true};
+  GPSData_t gpsData;
+  IMUData_t imuData;
```

Comment:
So I don't think you need to do this, what I mean is define all those data types individually and then set them to the PositionData_t struct on line 132.

To set the req_data on line 133, you can do `positionData.req_data.gps = true` then `positionData.req_data.imu = true`